runtime/defs.ts

```diff
@@ -142,6 +142,17 @@ export class Cons extends BaseAtom {
     }
   }
 
+  public some(func: (p: U) => boolean): boolean {
```

Comment:
This opens up a can of worms: if we add `some`, should we also add `map`, `filter`, `every` and other array-like methods.

alternative call-site syntax is `[...p].some(func)`

Reply:
I’m not hugely concerned… we might not need them all… or we might call them “someExpression” “mapExpression” (I’m improvising here) to avoid confusion if that’s the concern…

Reply:
I mean if we need something like “some” “filter” “map” that works on trees, we’ll find a good name, or we’ll just accept the homonyms as a fact of life. Unless you have a different concern that I don’t understand…

Reply:
As far as naming issues go, I’m more interested in finding a class hierarchy extending Us for taxonomysing interesting/special/common types of expressions, and giving better “alias” functions to car/cdr/etc. (say, for example “nextTerm” “nextFactor” etc.). Basically p.k and control flow based on p.k should eventually disappear and be subsumed by a class structure with the appropriate polymorphysms.

Reply:
That completely covers it and more. Thanks!

I like the priority of improving U taxonomy, naming, and removing `p.k` control-flow